api/api/controllers/search_controller.py

```diff
@@ -400,35 +360,78 @@ def search(
             query=f"{quotes_stripped}",
             boost=10000,
         )
-        s = search_client.query(Q("bool", must=s.query, should=exact_match_boost))
+        search_queries["should"].append(exact_match_boost)
     else:
-        if "creator" in search_params.data:
-            creator = _quote_escape(search_params.data["creator"])
-            s = s.query("simple_query_string", query=creator, fields=["creator"])
-        if "title" in search_params.data:
-            title = _quote_escape(search_params.data["title"])
-            s = s.query("simple_query_string", query=title, fields=["title"])
-        if "tags" in search_params.data:
-            tags = _quote_escape(search_params.data["tags"])
-            s = s.query("simple_query_string", fields=["tags.name"], query=tags)
+        for field, field_name in [
+            ("creator", "creator"),
+            ("title", "title"),
+            ("tags", "tags.name"),
+        ]:
+            if field_value := search_params.data.get(field):
+                search_queries["must"].append(
+                    Q(
+                        "simple_query_string",
+                        query=_quote_escape(field_value),
+                        fields=[field_name],
+                    )
+                )
 
     if settings.USE_RANK_FEATURES:
-        feature_boost = {"standardized_popularity": DEFAULT_BOOST}
-        if search_params.data["unstable__authority"]:
-            feature_boost["authority_boost"] = (
-                search_params.data["unstable__authority_boost"] * DEFAULT_BOOST
-            )
+        search_queries["should"].extend(create_ranking_queries(search_params))
+
+    # If there are no must query clauses, only the results that match
+    # the`should` clause are returned. To avoid this, we add an empty
+    # query clause to the `must` list.
```

Comment:
Thanks for adding this comment, I hadn't realised this distinction before or what `EMPTY_QUERY` was doing. What's the problem with only matching the should clauses? That should also go into the search algorithm documentation, I think.

```suggestion
    # If there are no `must` query clauses, only the results that match
    # the `should` clause are returned. To avoid this, we add an empty
    # query clause to the `must` list.
```

Reply:
I'm not fully certain why, but without `EMPTY_QUERY` in `must` clause, the search without parameters does not match any documents. In theory, all of the documents should match all of the `should` clauses.